atmat/lattice/atsimplering.m

```diff
@@ -0,0 +1,124 @@
+function ring = atsimplering(energy,circumference,hnum,...
+    qx,qy,vrf,alpha,varargin)
+%ATSIMPLERING   Creates a "simple ring"
+%
+% A "simple ring" consists of:
+% - A RF cavity
+% - a 6x6 linear transfer map
+% - a detuning and chromaticity element
+% - a simplified quantum diffusion element which provides the equilibrium
+% emittance, radiation damping and energy loss
+%
+%RING=ATSIMPLERING(ENERGY,CIRCUMFERENCE,HARMONIC_NUMBER,HTUNE,VTUNE,...
+%                  VRF,ALPHAC)
+% ENERGY:           nominal energy [eV]
+% CIRCUMFERENCE:    ring circumference [m]
+% HARMONIC_NUMBER:  harmonic number
+% HTUNE:            horizontal tune
+% VTUNE:            vertical tune
+% VRF:              RF voltage [V]
+% ALPHAC:           momentum compaction factor
+%
+% RING:             generated simple lattice
+%
+%RING=ATSIMPLERING(...,'name',name,...)
+%       specify the name of the generated lattice. Default: ''
+%
+%RING=ATSIMPLERING(...,'Particle',particle,...)
+%       specify the particle. Default: 'relativistic'
+%
+%RING=ATSIMPLERING(...,'Qpx',xsi_h,'Qpy',xsi_v,...)
+%       specify the chromaticites. Default: 0
+%
+%RING=ATSIMPLERING(...,'A1',A1,'A2',A2,'A3',A3,...)
+%       specify the amplitude detuning coefficients. Default: 0
+%
+%RING=ATSIMPLERING(...,'betax',betax,'betay',betay,...)
+%       specify the beta values at origin [m]. Default: 1
+%
+%RING=ATSIMPLERING(...,'betax',betax,'betay',betay,...)
+%       specify the alpha values at origin. Default: 0
+%
+%RING=ATSIMPLERING(...,'emitx',emitx,'emity',emity,...)
+%       specify the equilibrium emittance. Default: 0, meaning no quantum
+%       diffusion
+%
+%RING=ATSIMPLERING(...,'taux',taux,'tauy',tauy,'tauz',tauz,...)
+%       specify the damping times. Default: 0, meaning no damping
+%
+%RING=ATSIMPLERING(...,'U0',U0,...)
+%       specify the emergy loss per turn [eV]. Default: 0
+
+clight=PhysConstant.speed_of_light_in_vacuum.value;
+
+[particle,varargs]=getoption(varargin,'Particle',atparticle('relativistic'));
+[qpx,varargs]=getoption(varargs,'Qpx',0.0);
+[qpy,varargs]=getoption(varargs,'Qpy',0.0);
+[A1,varargs]=getoption(varargs,'A1',0.0);
+[A2,varargs]=getoption(varargs,'A2',0.0);
+[A3,varargs]=getoption(varargs,'A3',0.0);
+[taux,varargs]=getoption(varargs,'taux',0.0);
+[tauy,varargs]=getoption(varargs,'tauy',0.0);
+[tauz,varargs]=getoption(varargs,'tauz',0.0);
+[sigma,varargs]=getoption(varargs,'sigma',[]);
+[espread,varargs]=getoption(varargs,'Espread',0.0);
+if isempty(sigma)
+    [alphax,varargs]=getoption(varargs,'alphax',0.0);
+    [betax,varargs]=getoption(varargs,'betax',1.0);
+    [alphay,varargs]=getoption(varargs,'alphay',0.0);
+    [betay,varargs]=getoption(varargs,'betay',1.0);
+    [emitx,varargs]=getoption(varargs,'emitx',0.0);
+    [emity,varargs]=getoption(varargs,'emity',0.0);
```

Comment:
if I put in only the essential arguments for atsimplering, I get an unstable ring because the defaults here for emitx and emity are 0.0.  It gives an error at line 80.

Reply:
Right, if an emittance is zero, the optics cannot be deduced from sigma. That's corrected